### PR TITLE
fix #190

### LINKED ISSUE #190: check_for_200_error() AttributeError trying to set http_response.status_code
I hit this while uploading hundreds of thousands of files to S3 (full stack trace below):
```
  File "/home/kai/syncer/.env/lib/python3.6/site-packages/botocore/handlers.py", line 85, in check_for_200_error
    http_response.status_code = 500
AttributeError: can't set attribute
```

This is using 0.2.2.

```
aiobotocore==0.2.2
aiohttp==1.3.5
async-timeout==1.2.0
botocore==1.5.0
chardet==3.0.2
docutils==0.13.1
jmespath==0.9.2
multidict==2.1.4
python-dateutil==2.6.0
six==1.10.0
yarl==0.9.8
```

All I have is the stack trace, and I don't have any easy way to reproduce it.

```
Traceback (most recent call last):
  File "/home/kai/.pyenv/versions/3.6.1/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kai/.pyenv/versions/3.6.1/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/kai/syncer/syncer/__main__.py", line 53, in <module>
    loop.run_until_complete(main())
  File "/home/kai/.pyenv/versions/3.6.1/lib/python3.6/asyncio/base_events.py", line 466, in run_until_complete
    return future.result()
  File "/home/kai/syncer/syncer/__main__.py", line 47, in main
    await sm.sync_subtree(args.src_root, args.bucket, args.dest_root)
  File "/home/kai/syncer/syncer/syncmaster.py", line 80, in sync_subtree
    lambda a: self.validate_or_upload(*a))))
  File "/home/kai/syncer/syncer/atools.py", line 23, in ajunk
    async for _ in aseq:
  File "/home/kai/syncer/syncer/syncmaster.py", line 343, in amap_with_flow
    yield await task
  File "/home/kai/syncer/syncer/syncmaster.py", line 247, in upload
    Parts=complete_parts,
  File "/home/kai/syncer/.env/lib/python3.6/site-packages/aiobotocore/client.py", line 78, in _make_api_call
    operation_model, request_dict)
  File "/home/kai/syncer/.env/lib/python3.6/site-packages/aiobotocore/endpoint.py", line 214, in _send_request
    exception)):
  File "/home/kai/syncer/.env/lib/python3.6/site-packages/aiobotocore/endpoint.py", line 246, in _needs_retry
    caught_exception=caught_exception, request_dict=request_dict)
  File "/home/kai/syncer/.env/lib/python3.6/site-packages/botocore/hooks.py", line 227, in emit
    return self._emit(event_name, kwargs)
  File "/home/kai/syncer/.env/lib/python3.6/site-packages/botocore/hooks.py", line 210, in _emit
    response = handler(**kwargs)
  File "/home/kai/syncer/.env/lib/python3.6/site-packages/botocore/handlers.py", line 85, in check_for_200_error
    http_response.status_code = 500
AttributeError: can't set attribute
```

### PR DESCRIPTION
**Pull request by thehesiod**
fix #190

### CODE REVIEW

**Comment by jettify** [comment]
Sorry missed this PR. Lets add simple test case something like:
```python
resp = ClientResponseProxy(status=200)
check_for_200_error(res)
asser res.status_code ==500
```
just to document with test that response should be settable.

**Comment by thehesiod** [comment]
me too :)

File: CHANGES.txt [hunk 1]
```diff
@@ -15,6 +15,7 @@ Changes
 0.4.4 (2017-08-16)
 ^^^^^^^^^^^^^^^^^^
 * add the supported versions of boto3 to extras require #324
+>>>>>>> master
```
Comment (jettify) [inline CHANGES.txt]: probably leftover after merge

**Comment by jettify** [comment]
Lets wait a few days just in case something pops up, and then make release.

**Pull request merged**